tests/ha/test_multi_server_pod_failure.py

```diff
@@ -1880,3 +1879,578 @@ def test_chunk_upload_during_kserver_pods_down(self):
+            self.restore_pod = self.deploy = True
+            self.restore_method = const.RESTORE_DEPLOYMENT_K8S
+            self.pod_dict[pod_name] = pod_data
+            LOGGER.info("Deleted %s server pod %s by deleting deployment"
+                        " (unsafe)", count, pod_name)
+        LOGGER.info("Step 3: Successfully shutdown %s (K) server pods one by one while "
+                    "continuous DELETEs in background", self.kvalue)
+
+        LOGGER.info("Step 4: Check cluster status")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        assert_utils.assert_false(resp[0], resp)
+        LOGGER.info("Step 4: Cluster has some failures due to %s server pods has gone down.",
+                    self.kvalue)
+
+        LOGGER.info("Step 5: Check services status that were running on server "
+                    "pods which are deleted.")
+        counter = 0
+        for pod_name in self.pod_name_list:
+            hostname = self.pod_dict.get(pod_name)[0]
+            resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=[pod_name], fail=True,
+                                                               hostname=hostname)
+            LOGGER.debug("Services status on %s : %s", pod_name, resp)
+            if not resp[0]:
+                counter += 1
+            server_data_pod_list.remove(pod_name)
+        assert_utils.assert_equal(counter, 0, "Services on some server pods not stopped.")
+        LOGGER.info("Step 5: Services of deleted server pods are in offline state")
+
+        LOGGER.info("Step 6: Check services status on remaining pods %s", server_data_pod_list)
+        resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=server_data_pod_list,
+                                                           fail=False)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 6: Services of remaining pods are in online state")
+        event.clear()
+
+        LOGGER.info("Step 7: Verify status for In-flight DELETEs while %s (K) server pods were"
+                    "going down", self.kvalue)
+        LOGGER.info("Waiting for background DELETEs thread to join. Waiting for %s seconds to "
+                    "collect the queue logs", HA_CFG["common_params"]["60sec_delay"])
+        thread.join()
+        del_resp = tuple()
+        while len(del_resp) != 2:
+            del_resp = del_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not del_resp:
+            assert_utils.assert_true(False, "Background process failed to do deletes")
+        event_del_bkt = del_resp[0]
+        fail_del_bkt = del_resp[1]
+        rem_bkts_aftr_del = s3_test_obj.bucket_list()[1]
+        assert_utils.assert_false(len(fail_del_bkt),
+                                  f"Bucket deletion failed when cluster was online {fail_del_bkt}")
+        assert_utils.assert_true(len(rem_bkts_aftr_del) < del_bucket,
```

Comment:
rather we should check len(event_del_bkt) != 0, and say some bucket deletion was expected.
Otherwise, if its getting stuck or someting gets wrong, it will pass as empty list to next put_get_delete()
@ApurwaMohite  @sonalk0209 suggestion Please

Reply:
Yes, this check can be added or we can add remaining_bkt < len(rem_bkts_aftr_del) < del_bucket.

Reply:
but ankita we are checking at next step : len(event_del_bkt) ? so is this really needed?

Reply:
This is part of step 6. right

Reply:
ok will add this change.

Reply:
@rahul-hatwar  as this is server pod, len(event_del_bkt) should be 0 always.
And step 8 should not be needed.